source/xep/httpFileUpload/DownloadManager.cpp

```diff
@@ -147,3 +189,58 @@ void DownloadManager::downloadFinished(QNetworkReply *reply)
     }
 }
 
+
+bool DownloadManager::aesDecrypt(const QByteArray &key, const QByteArray &iv, const QByteArray &in, QByteArray &out)
+{
+    gcry_cipher_hd_t cipher_hd = 0;
+    gcry_error_t ret_val;
+
+    ret_val = gcry_cipher_open(&cipher_hd, GCRY_CIPHER_AES256, GCRY_CIPHER_MODE_CTR, 0);
+
+    if(ret_val) {
+        qDebug() << "failed to initialize gcrypt";
+        goto cleanup;
+    }
+
+    if(key.count() != 32) 
+    {
+        qDebug() << "invalid key size: " << key.count();
+        goto cleanup;
+    }
+
+    if(iv.count() != 16) 
+    {
+        qDebug() << "invalid iv size: " << iv.count();
+        goto cleanup;
+
+    }
+
+    ret_val = gcry_cipher_setkey(cipher_hd, key.constData(), key.size());
+    if (ret_val) {
+        qDebug() << "failed to set key";
+        goto cleanup;
+    }
+
+    ret_val = gcry_cipher_setctr(cipher_hd, iv.constData(), iv.size());
```

Comment:
Compiler warning: implizit conversion...

Reply:
I don't have the warning and I don't see it

Reply:
Hm, interesting. So if you prefer I could work on the compiler warnings by myself and accept the PR as it is now. Does this work for you?